tests/ssd_r34_relay_to_glenside.rs

```diff
@@ -0,0 +1,59 @@
+#![cfg(feature = "tvm")]
+
+use egg::EGraph;
+use glenside::language::MyAnalysis;
+use std::collections::HashMap;
+use std::path::PathBuf;
+
+// Mobilenet, simplified for inference (so batch norms are simplified).
+// Generate with:
+// ```python3
+// import tvm
+// from tvm import relay
+// from tvm.relay.testing.mobilenet import get_workload
+//
+// mod, _ = get_workload()
+// mod = relay.transform.SimplifyInference()(mod)
+// print(mod.astext())
+// ```
+
+// Mobilenet
+// Generate with:
+// ```python3
+// import tvm
+// from tvm import relay
+// from tvm.relay.testing.mobilenet import get_workload
+//
+// mod, _ = get_workload()
+// print(mod.astext())
+// ```
+#[test]
+fn parse_ssd_r34() {
+    let filename = PathBuf::from(format!(
+        "{}/models/ssd_r34.relay",
+        env!("CARGO_MANIFEST_DIR")
+    ));
+    let relay = std::fs::read_to_string(&filename).unwrap();
+    let module = tvm::ir::module::IRModule::parse("", relay).unwrap();
+
+    let (expr, shapes_vec, dtypes_vec) = glenside::language::from_relay::from_relay(
+        &module,
+        true,
+        &glenside::language::RELAY_OPS.into(),
+    );
+
+    let mut env = HashMap::default();
+    for (k, v) in &shapes_vec {
+        env.insert(k.clone(), v.clone());
+    }
+
+    // TODO(@gussmith23) Include some simple simplifying rewrites
+    // If we add some very basic rewrites here, then $glenside_str
+    // won't need to exactly match what's actually produced by
+    // from_relay.py. It can be simpler (e.g. collapsing accesses).
+    let mut egraph = EGraph::new(MyAnalysis {
+        name_to_shape: env.clone(),
+        name_to_dtype: dtypes_vec.into_iter().collect(),
+    });
+    egraph.add_expr(&expr);
```

Comment:
Can you add a check here, to ensure that Glenside's type system infers the same output shape as Relay's type system? Rough sketch on how to do this:
1. Put the `Id` returned by `add_expr` into a variable (`let id = egraph.add_expr(...);`). This is the id of the expression representing the model in the egraph.
2. Get the `MyAnalysisData` object for the expression: `egraph[id].data`. This object contains the analysis data for the eclass containing our SSD R34 expression. The part of the analysis data that we care about is the shape information.
3. Check that the `MyAnalysisData` looks like we expect. Namely, check that the inferred shape of the expression is the same as the shape that Relay infers (i.e. the return type of the `main` function in the Relay code!)

You can look at this code for inspiration:
https://github.com/gussmith23/glenside/blob/cfb4f9cc12607ff1990cfebe45ded11fecd5912f/tests/transformer.rs#L38-L46

A key difference is that it looks like SSD R34 will return a tuple of two access patterns, rather than returning a single access pattern. This means that when you match on `egraph[id].data`, you will not expect a `MyAnalysisData::AccessPattern`, but a `MyAnalysisData::Tuple`.

Reply:
done